cloud/pkg/synccontroller/objectsync.go

```diff
@@ -203,15 +203,20 @@ func GetObjectResourceVersion(obj interface{}) string {
 // ints, so we can easily compare them.
 // If rva>rvb, return 1; rva=rvb, return 0; rva<rvb, return -1
 func CompareResourceVersion(rva, rvb string) int {
-	a, err := strconv.ParseUint(rva, 10, 64)
-	if err != nil {
-		// coder error
-		panic(err)
+	var a, b uint64
+	var err error
+
+	if len(rva) > 0 {
+		a, err = strconv.ParseUint(rva, 10, 64)
+		if err != nil {
+			klog.Warningf("parsing ResourceVersion a failed with: %s", err)
+		}
 	}
-	b, err := strconv.ParseUint(rvb, 10, 64)
-	if err != nil {
-		// coder error
-		panic(err)
+	if len(rvb) > 0 {
+		b, err = strconv.ParseUint(rvb, 10, 64)
+		if err != nil {
+			klog.Warningf("parsing ResourceVersion b failed with: %s", err)
+		}
```

Comment:
Thanks your quick fix!
If rva or rvb is empty, I feel like there's no need to do the Comparation and send message. We can add the judgment at [L161](https://github.com/kubeedge/kubeedge/pull/1529/files#diff-a6dde63cc14b4596a794b990dacdddcdR161) instead. @alesz

Reply:
Seems like behaviour is caused by getObjectResourceVersion returning empty string when called on edge side resource. Modified to always return valid value as string to allow parsing and simpler update trigger check.

Reply:
Panic happened in [objectsync.go#L214](https://github.com/kubeedge/kubeedge/blob/master/cloud/pkg/synccontroller/objectsync.go#L214), since the `syncResourceVersion` is empty, and in this case, events should not be sent.